dataclients/kubernetes/resources.go

```diff
@@ -24,3 +24,10 @@ type secret struct {
 type secretList struct {
 	Items []*secret `json:"items"`
 }
+
+type objectReference struct {
+	Kind      string `json:"kind"`
+	Name      string `json:"name"`
```

Comment:
Name is the only thing we need, rest is either:
- static: Kind is always "Pod"
- duplicate from endpointslice object: Namespace
- useless: Uid

Reply:
I don't want that we pass kubernetes structs. Pass what we need and no references here.

Reply:
Please read my comment, this struct is to unmarshal data from k8s API. We can put subset of data into metadata route, I'll update that later.

Reply:
> I'll update that later.

Updated to not use objectReference in the encoded metadata